Description: Disable code that relies on patented software.
 Virtually all free-software distributions cannot support MPEG2, since it is
 patented -- hence VTK comes without it. This patch introduced a switch to turn
 it off and removes traces from the build setup. This also disables all pieces
 that rely on patented code that is shipped by upstream, but got remove by
 repacking for Debian.
Author: Michael Hanke <michael.hanke@gmail.com>
Last-Update: 2010-04-08
--- a/caret/GuiRecordingDialog.cxx
+++ b/caret/GuiRecordingDialog.cxx
@@ -62,8 +62,10 @@
 #ifdef HAVE_AVI_WRITER
 #include "vtkAVIWriter.h"
 #endif // HAVE_AVI_WRITER
+#ifdef HAVE_VTK_MPEG
 #include "vtkMPEG1Writer.h"
 #include "vtkMPEG2Writer.h"
+#endif // HAVE_VTK_MPEG
 #endif // HAVE_VTK5
 
 #include "BrainModelRunExternalProgram.h"
@@ -169,8 +171,10 @@
    dialogLayout->addLayout(buttonsLayout);
    
 #ifdef HAVE_VTK5
+#ifdef HAVE_VTK_MPEG
    vtkMpeg1MovieWriter = NULL;
    vtkMpeg2MovieWriter = NULL;
+#endif // HAVE_VTK_MPEG
 #ifdef HAVE_AVI_WRITER
    vtkAviMovieWriter = NULL;
 #endif // HAVE_AVI_WRITER
@@ -185,6 +189,7 @@
 GuiRecordingDialog::~GuiRecordingDialog()
 {
 #ifdef HAVE_VTK5
+#ifdef HAVE_VTK_MPEG
    if (vtkMpeg2MovieWriter != NULL) {
       vtkMpeg2MovieWriter->Delete();
       vtkMpeg2MovieWriter = NULL;
@@ -193,6 +198,7 @@
       vtkMpeg1MovieWriter->Delete();
       vtkMpeg1MovieWriter = NULL;
    }
+#endif // HAVE_VTK_MPEG
 #ifdef HAVE_AVI_WRITER
    if (vtkAviMovieWriter != NULL) {
       vtkAviMovieWriter->Delete();
@@ -421,16 +427,20 @@
    movieFileTypeComboBox->addItem("MPEG (using mpeg_create)", 
                                   static_cast<int>(MOVIE_FILE_TYPE_MPEG_CREATE));
 #ifdef HAVE_VTK5
+#ifdef HAVE_VTK_MPEG
    movieFileTypeComboBox->addItem("MPEG1 (using VTK)", 
                                   static_cast<int>(MOVIE_FILE_TYPE_MPEG1_VTK));
    const int mpeg1Index = movieFileTypeComboBox->count() - 1;
    movieFileTypeComboBox->addItem("MPEG2 (using VTK)", 
                                   static_cast<int>(MOVIE_FILE_TYPE_MPEG2_VTK));
+#endif // HAVE_VTK_MPEG
 #ifdef HAVE_AVI_WRITER
    movieFileTypeComboBox->addItem("AVI (using VTK)", 
                                   static_cast<int>(MOVIE_FILE_TYPE_AVI_VTK));
 #endif // HAVE_AVI_WRITER
+#ifdef HAVE_VTK_MPEG
    movieFileTypeComboBox->setCurrentIndex(mpeg1Index);
+#endif // HAVE_VTK_MPEG
 #endif // HAVE_VTK5
    movieFileTypeComboBox->addItem("JPEG Images",
                                   static_cast<int>(MOVIE_FILE_TYPE_JPEG_IMAGES));
@@ -575,6 +585,7 @@
 GuiRecordingDialog::createMovieWithMpeg1VTK()
 {
 #ifdef HAVE_VTK5
+#ifdef HAVE_VTK_MPEG
    //
    // Finish the movie
    //
@@ -588,6 +599,7 @@
    // Cannot append images so delete all captured images
    //
    deleteTemporaryImages();
+#endif // HAVE_VTK_MPEG
 #endif // HAVE_VTK5
 }
 
@@ -598,6 +610,7 @@
 GuiRecordingDialog::createMovieWithMpeg2VTK()
 {
 #ifdef HAVE_VTK5
+#ifdef HAVE_VTK_MPEG
    //
    // Finish the movie
    //
@@ -611,6 +624,7 @@
    // Cannot append images so delete all captured images
    //
    deleteTemporaryImages();
+#endif // HAVE_VTK_MPEG
 #endif // HAVE_VTK5
 }
 
@@ -883,6 +897,7 @@
 GuiRecordingDialog::addImageToMpeg1VTK(const QImage& image)
 {
 #ifdef HAVE_VTK5
+#ifdef HAVE_VTK_MPEG
    const QString fileName(movieNameLineEdit->text());
    vtkImageData* vtkImage = convertQImagetoVTKImageData(image);
    if (vtkMpeg1MovieWriter == NULL) {
@@ -900,6 +915,7 @@
    if (DebugControl::getDebugOn()) {
       std::cout << "MPEG Writer Error Code: " << vtkMpeg1MovieWriter->GetErrorCode() << std::endl;
    }
+#endif // HAVE_VTK_MPEG
 #endif // HAVE_VTK5
 }
 
@@ -910,6 +926,7 @@
 GuiRecordingDialog::addImageToMpeg2VTK(const QImage& image)
 {
 #ifdef HAVE_VTK5
+#ifdef HAVE_VTK_MPEG
    const QString fileName(movieNameLineEdit->text());
    vtkImageData* vtkImage = convertQImagetoVTKImageData(image);
    if (vtkMpeg2MovieWriter == NULL) {
@@ -927,6 +944,7 @@
    if (DebugControl::getDebugOn()) {
       std::cout << "MPEG Writer Error Code: " << vtkMpeg2MovieWriter->GetErrorCode() << std::endl;
    }
+#endif // HAVE_VTK_MPEG
 #endif // HAVE_VTK5
 }
 
@@ -1121,6 +1139,7 @@
 GuiRecordingDialog::deleteTemporaryImages()
 {
 #ifdef HAVE_VTK5
+#ifdef HAVE_VTK_MPEG
    if (vtkMpeg1MovieWriter != NULL) {
       vtkMpeg1MovieWriter->Delete();
       vtkMpeg1MovieWriter = NULL;
@@ -1129,6 +1148,7 @@
       vtkMpeg2MovieWriter->Delete();
       vtkMpeg2MovieWriter = NULL;
    }
+#endif // HAVE_VTK_MPEG
 #ifdef HAVE_AVI_WRITER
    if (vtkAviMovieWriter != NULL) {
       vtkAviMovieWriter->Delete();
--- a/caret/GuiRecordingDialog.h
+++ b/caret/GuiRecordingDialog.h
@@ -43,8 +43,10 @@
 
 class vtkAVIWriter;
 class vtkImageData;
+#ifdef HAVE_VTK_MPEG
 class vtkMPEG1Writer;
 class vtkMPEG2Writer;
+#endif // HAVE_VTK_MPEG
 
 /// Dialog for recording to movie file.
 class GuiRecordingDialog : public WuQDialog {
@@ -222,11 +224,13 @@
       /// pointer for VTK's AVI writer
       vtkAVIWriter* vtkAviMovieWriter;
       
+#ifdef HAVE_VTK_MPEG
       /// pointer for caret's modified VTK's MPEG2 writer that writes MPEG1
       vtkMPEG1Writer* vtkMpeg1MovieWriter;
       
       /// pointer for VTK's MPEG2 writer
       vtkMPEG2Writer* vtkMpeg2MovieWriter;
+#endif // HAVE_VTK_MPEG
 };
 
 
--- a/caret_command/caret_command.pro
+++ b/caret_command/caret_command.pro
@@ -82,7 +82,6 @@
    PRE_TARGETDEPS +=  \
            ../caret_command_operations/libCaretCommandOperations.a \
            ../caret_brain_set/libCaretBrainSet.a \
-           ../caret_vtk4_classes/libCaretVtk4Classes.a \
            ../caret_files/libCaretFiles.a \
            ../caret_widgets/libCaretWidgets.a \
            ../caret_uniformize/libCaretUniformize.a \
@@ -96,7 +95,6 @@
    LIBS += \
            -L../caret_command_operations -lCaretCommandOperations \
            -L../caret_brain_set -lCaretBrainSet \
-           -L../caret_vtk4_classes -lCaretVtk4Classes \
            -L../caret_files -lCaretFiles \
            -L../caret_widgets -lCaretWidgets \
 	        -L../caret_uniformize -lCaretUniformize \
--- a/caret_qmake_include.pro
+++ b/caret_qmake_include.pro
@@ -129,7 +129,9 @@
 #
 # Check for VTK 5.x
 #
-exists( $(VTK_INC_DIR)/vtkMPEG2Writer.h ) {
+# MPEG2 code is not shipped with many distribution versions of VTK
+#exists( $(VTK_INC_DIR)/vtkMPEG2Writer.h ) {
+exists( $(VTK_INC_DIR)/vtkImageSpatialAlgorithm.h ) {
   
    message( "Building WITH VTK5 support" )
 
@@ -141,12 +143,12 @@
    #
    #
    #
-   INCLUDEPATH += ../caret_vtk4_classes
+#   INCLUDEPATH += ../caret_vtk4_classes
 
    #
    # need by caret_vtk4_classes
    #
-   INCLUDEPATH += $(VTK_INC_DIR)/vtkmpeg2encode
+#   INCLUDEPATH += $(VTK_INC_DIR)/vtkmpeg2encode
 
    #
    # VTK Libraries for VTK 5.x
@@ -155,7 +157,7 @@
       #VTK_LIBS = ../caret_vtk4_classes/debug/libCaretVtk4Classes.a 
    }
    !win32 {
-      VTK_LIBS = ../caret_vtk4_classes/libCaretVtk4Classes.a 
+#      VTK_LIBS = ../caret_vtk4_classes/libCaretVtk4Classes.a
    }
    VTK_LIBS += \
               -L$(VTK_LIB_DIR) \
@@ -165,7 +167,6 @@
               -lvtkImaging \
               -lvtkGraphics \
               -lvtkIO \
-              -lvtkMPEG2Encode \
               -lvtkFiltering \
               -lvtkCommon \
               -lvtkjpeg \
@@ -177,7 +178,9 @@
 #
 # Check for VTK 4.x (does not have vtkMPEG2Writer.h)
 #
-!exists( $(VTK_INC_DIR)/vtkMPEG2Writer.h ) {
+# MPEG2 code is not shipped with many distribution versions of VTK
+#!exists( $(VTK_INC_DIR)/vtkMPEG2Writer.h ) {
+!exists( $(VTK_INC_DIR)/vtkImageSpatialAlgorithm.h ) {
    #
    # VTK Libraries for VTK 4.x
    #
--- a/caret/caret5.pro
+++ b/caret/caret5.pro
@@ -104,7 +104,6 @@
    PRE_TARGETDEPS +=  \
       ../caret_command_operations/libCaretCommandOperations.a \
 	   ../caret_brain_set/libCaretBrainSet.a \
-      ../caret_vtk4_classes/libCaretVtk4Classes.a \
 	   ../caret_files/libCaretFiles.a \
 	   ../caret_uniformize/libCaretUniformize.a \
 	   ../caret_statistics/libCaretStatistics.a \
@@ -116,7 +115,6 @@
    LIBS +=  \
       -L../caret_command_operations -lCaretCommandOperations \
 	   -L../caret_brain_set -lCaretBrainSet \
-      -L../caret_vtk4_classes -lCaretVtk4Classes \
 	   -L../caret_files -lCaretFiles \
 	   -L../caret_uniformize -lCaretUniformize \
 	   -L../caret_statistics -lCaretStatistics \
